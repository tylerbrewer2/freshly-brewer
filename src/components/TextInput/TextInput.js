import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

// Style Helpers
import Fonts from '../../styles/helpers/Fonts';

const TextInputCore = ({ className, identifier, onChange, ...rest }) => {
  return (
    <input
      type='text'
      className={className}
      name={identifier}
      id={identifier}
      onChange={onChange}
      {...rest}
    />
  )
}

TextInputCore.propTypes = {
  className: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired,
  identifier: PropTypes.string,
}

TextInputCore.defaultProps = {
  identifier: null,
}

const TextInput = styled(TextInputCore)`
  font-family: ${Fonts.body};
  font-size: 18px;
  width: 100%;
  border: 1px solid #4A4A4A;
  border-radius: 3px;
  padding: 10px;
`;

export default TextInput;
