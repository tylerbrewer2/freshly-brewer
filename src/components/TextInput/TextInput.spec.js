import React from 'react';
import renderer from 'react-test-renderer';

import TextInput from './TextInput';

describe('component structure', () => {
  it('renders correctly', () => {
    const tree = renderer.create(
      <TextInput
        status="success"
        identifier="test"
        onChange={jest.fn()}
        value='Test'
      />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
