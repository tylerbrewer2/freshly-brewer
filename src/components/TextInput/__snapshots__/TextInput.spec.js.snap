exports[`component structure renders correctly 1`] = `
<input
  className="sc-bdVaJa eFURHh"
  id="test"
  name="test"
  onChange={[Function]}
  status="success"
  type="text"
  value="Test" />
`;
